src/telemetry/logs.ts

```diff
@@ -53,30 +70,64 @@ export class GlobalLogger {
     );
     // Only enable the OTLP transport if we have a telemetry collector and an exporter
     if (this.telemetryCollector?.exporter) {
-      winstonTransports.push(new OTLPLogQueueTransport(this.telemetryCollector, config?.logLevel || 'info'));
+      this.otlpTransport = new OTLPLogQueueTransport(this.telemetryCollector, config?.logLevel || 'info');
+      winstonTransports.push(this.otlpTransport);
     }
     this.logger = createLogger({ transports: winstonTransports });
     this.addContextMetadata = config?.addContextMetadata || false;
+
+    if (process.env.DBOS__CAPTURE_STD !== 'false') {
+      this.captureStdoutAndStderr();
+    }
+  }
+
+  private captureStdoutAndStderr(): void {
+    // First flush the buffers to the OTLP transport
+    if (this.telemetryCollector?.exporter) {
+      for (const line of logBuffer.flush()) {
+        this.otlpTransport?.log({ level: 'info', message: line.trim() }, () => {});
+      }
+      for (const line of logErrorBuffer.flush()) {
+        this.otlpTransport?.log({ level: 'error', message: line.trim() }, () => {});
+      }
+    }
+    // Overwrite the intercepts again to capture non-DBOS logger stdout/stderr and export them through OTLP
+    interceptStreams((msg, stream) => {
+      if (stream === 'stdout') {
+        if (!this.isLogging) {
+          this.otlpTransport?.log({ level: 'info', message: msg.trim() }, () => {});
+        }
+      } else {
+        if (!this.isLogging) {
+          this.otlpTransport?.log({ level: 'error', message: msg.trim(), stack: new Error().stack }, () => {});
+        }
+      }
+    });
   }
 
   // We use this form of winston logging methods: `(message: string, ...meta: any[])`. See node_modules/winston/index.d.ts
   info(logEntry: unknown, metadata?: ContextualMetadata): void {
+    this.isLogging = true;
     if (typeof logEntry === 'string') {
       this.logger.info(logEntry, metadata);
     } else {
       this.logger.info(DBOSJSON.stringify(logEntry), metadata);
     }
+    this.isLogging = false;
   }
 
   debug(logEntry: unknown, metadata?: ContextualMetadata): void {
+    this.isLogging = true;
     if (typeof logEntry === 'string') {
       this.logger.debug(logEntry, metadata);
     } else {
       this.logger.debug(DBOSJSON.stringify(logEntry), metadata);
     }
+    this.isLogging = false;
   }
 
   warn(logEntry: unknown, metadata?: ContextualMetadata): void {
+    this.isLogging = true;
```

Comment:
There's no corresponding `false` here. Is there a safer way to do this?

Reply:
A simpler and safer solution is to prepend all our log entries with `[DBOS]` and, in the override, do the OTLP export  `if (!msg.startsWith('[DBOS]'))`